cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewStatus.java

```diff
@@ -33,7 +33,8 @@
     COMPLETED,
     DEPLOY_FAILED,
     RUN_FAILED,
-    KILLED
+    KILLED,
+    KILLED_BY_TIMER
```

Comment:
I dont think we kill the preview based on the number of records processed. And I do not see a way about killing the pipeline after we process a certain number of records